src/Streamly/Internal/Data/Stream/IsStream/Expand.hs

```diff
@@ -704,6 +705,14 @@ concatPairsWith :: IsStream t =>
     -> t m b
 concatPairsWith = K.concatPairsWith
 
+{-# INLINE concatPairsWithD #-}
+concatPairsWithD :: (IsStream t , Monad m) =>
+       (S.Stream m b -> S.Stream m b -> S.Stream m b)
+    -> (a -> S.Stream m b)
+    -> t m a
+    -> t m b
```

Comment:
I have a few problems with this.

Any kind of combination function is more efficient with K-ish
combinators.

Users using IsStream should not be able to know what `StreamD`
is. I believe the high level picture is to abstract it out. Or
even if it isn't I don't see where this can be used.

We can maybe have a `concatPairsWith` in `StreamD` itself and use
`fromStreamD` to lift it here.

Then the signature would be similar to `concatPairsWith`. But
then again, I don't see how and where that would be needed.